File: src/main/java/com/technews/aggregate/releases/service/ReleasesSchedulerService.java
package com.technews.aggregate.releases.service;

import com.technews.aggregate.releases.domain.Release;
import com.technews.aggregate.releases.domain.repository.ReleasesRepository;
import com.technews.aggregate.releases.dto.SaveReleaseRequest;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Slf4j
@Service
@RequiredArgsConstructor
public class ReleasesSchedulerService {

    private final ReleasesRepository releasesRepository;

    @Transactional
    public void insertRelease(SaveReleaseRequest saveReleaseRequest) {
        try {
            releasesRepository.save(saveReleaseRequest.toRelease());
            log.info("add new release version. {}", saveReleaseRequest.getVersion());
        } catch (Exception e) {
            log.error("ReleasesSchedulerService.insertRelease exception", e);
        }
    }

    @Transactional(readOnly = true)
    public Release findLatestRelease(final String project) {
        final List<Release> latestReleaseDate = releasesRepository.findByProjectOrderByVersionDescLimitOne(project);
        if (latestReleaseDate.isEmpty()) {
            return Release.EMPTY;
        }

        return latestReleaseDate.get(0);
    }
}

File: src/main/kotlin/com/technews/aggregate/releases/service/ReleasesSchedulerService.kt
package com.technews.aggregate.releases.service

import com.technews.aggregate.releases.domain.Release
import com.technews.aggregate.releases.domain.repository.ReleasesRepository
import com.technews.aggregate.releases.dto.SaveReleaseRequest
import mu.KotlinLogging
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

private val logger = KotlinLogging.logger {}

@Service
class ReleasesSchedulerService(
    private val releasesRepository: ReleasesRepository,
) {
    @Transactional
    fun insertRelease(saveReleaseRequest: SaveReleaseRequest) {
        runCatching {
            releasesRepository.save<Release>(saveReleaseRequest.toRelease())
            logger.info("add new release version. ${saveReleaseRequest.version}")
        }.onFailure { e ->
            logger.error("ReleasesSchedulerService.insertRelease exception", e)
        }
    }

    @Transactional(readOnly = true)
    fun findLatestRelease(project: String): Release =
        releasesRepository.findByProjectOrderByVersionDescLimitOne(project).firstOrNull() ?: Release()
}
